crates/protocol/src/batch/transactions.rs

```diff
@@ -260,11 +260,8 @@ impl SpanBatchTransactions {
                 .tx_sigs
                 .get(idx as usize)
                 .ok_or(SpanBatchError::Decoding(SpanDecodingError::InvalidTransactionData))?;
-            let is_protected = self
-                .protected_bits
-                .get_bit(idx as usize)
-                .ok_or(SpanBatchError::Decoding(SpanDecodingError::InvalidTransactionData))?
-                == 1;
+            let is_protected =
+                self.protected_bits.get_bit(idx as usize).unwrap_or(sig.v() as u8) == 1;
```

Comment:
@klkvr in your alloy update pr, span batch txs decoding broke when a tx didn't have protected bits - I've updated the `SpanBatchTransactions` handling here to use the signature v value if the protected bits are missing. This fixes batch reading. I've upstreamed the `BatchReader` from kona as well in this PR which validates batch reading - prior to this pr, decoding a valid batch with the `BatchReader` failed.

Reply:
For reference: https://github.com/anton-rs/kona/pull/786

Reply:
looks like before `None` was treated as false already

https://github.com/alloy-rs/op-alloy/blob/edc82479bc6b0d2f48176f278638eaedf892b5f9/crates/protocol/src/batch/transactions.rs#L247-L250

Reply:
Yeah if the protected bit isn't found, we should be treating the signature as unprotected.